### PR TITLE
Adding ActionResult info into ActionCardComponent

### PR DESCRIPTION
**Pull request by pariparajuli**
Formatted ActionCardComponent with Action Result included

### CODE REVIEW

**Comment by pariparajuli** [comment]
Not sure why the build is failing at the bytestream server, it seems to work on my local machine

**Comment by pariparajuli** [comment]
[
Screen Shot 2021-06-01 at 2 49 00 PM
](url)

**Comment by pariparajuli** [comment]
Screen Shot 2021-06-01 at 2 55 30 PM
the existing formatting functions don't go to the miliseconds so i'm tinkering with that but this is what it looks like now

File: app/invocation/invocation_execution_card.tsx [hunk 1]
```diff
@@ -290,7 +290,7 @@ export default class ExecutionCardComponent extends React.Component {
                 <div>
                   <div
                     className="invocation-execution-row-digest"
-                    onClick={capabilities.action ? this.handleActionDigestClick.bind(this, execution) : undefined}>
+                    onClick={this.handleActionDigestClick.bind(this, execution)}>
```
Comment (siggisim) [inline app/invocation/invocation_execution_card.tsx:293]: do we want to wait to enable this until we've got the localhost stuff figured out?

File: app/invocation/invocation_action_card.tsx [hunk 2]
```diff
@@ -127,65 +128,126 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                      <div className="action-property-title">Execution Metadata:</div>
+                      {this.state.actionResult.executionMetadata ? (
+                        <div className="action-list">
+                          <div className="metadata-title">Worker</div>
+                          <div className="metadata-detail">{this.state.actionResult.executionMetadata.worker} </div>
+                          <div className="metadata-title">Executor ID</div>
+                          <div className="metadata-detail">{this.state.actionResult.executionMetadata.executorId}</div>
+                          <div className="metadata-title">Timeline</div>
+                          <div className="metadata-detail">
+                            Queued @ {this.state.actionResult.executionMetadata.queuedTimestamp.nanos}
```
Comment (siggisim) [inline app/invocation/invocation_action_card.tsx:176]: you should just be able to add another formatter here that supports nanos. totally cool if you want to do that in a separate PR

https://github.com/buildbuddy-io/buildbuddy/blob/master/app/format/format.tsx#L14

**Review by siggisim — COMMENTED** [review]
looks good, just a couple of nits

**Comment by pariparajuli** [comment]
Screen Shot 2021-06-02 at 2 22 56 PM

File: app/invocation/invocation_action_card.tsx [hunk 3]
```diff
@@ -127,65 +128,155 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                          {this.state.actionResult.outputFiles.map((file) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
+                              <span className="prop-value">{file.path}</span>
+                              <div>
+                                <span className="prop-name">Executable? </span>
+                                <span className="prop-value">{file.isExecutable.toString().toUpperCase()}</span>
+                              </div>
+                            </div>
+                          ))}
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:256]: Could you format this as just the path name, followed by (executable) in parens if it's executable?

So like:

`foo/bar/baz`
`foo/bar (executable)`

Maybe put the "(executable)" part in lighter grey so it's clear that "(executable)" is not part of the path name.

File: app/invocation/invocation_action_card.tsx [hunk 4]
```diff
@@ -127,65 +128,155 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
+                      <div className="action-property-title">Output Directories:</div>
+                      {this.state.actionResult.outputDirectories.length ? (
+                        <div className="action-list">
+                          {this.state.actionResult.outputDirectories.map((file) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:268]: Same comment here re. path formatting

File: app/invocation/invocation_action_card.tsx [hunk 5]
```diff
@@ -127,65 +128,155 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.seconds,
+                              this.state.actionResult.executionMetadata.workerCompletedTimestamp.nanos
+                            )}
+                          </div>
+                        </div>
+                      ) : (
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:243]: Commands also have an input directory digest -- we might not want to fetch the complete tree of inputs and show it in the UI in this PR, but maybe we could at least show the total size of inputs just above this? (You could get that as the size field of the input root digest field)

**Review by bduffany — CHANGES_REQUESTED** [review]

File: app/invocation/invocation_action_card.tsx (hunk 4 above)
Comment (pariparajuli) [reply app/invocation/invocation_action_card.tsx:270]: ah these are directories that don't have the executable information, i renamed file to dir for less confusion

File: app/invocation/invocation_action_card.tsx [hunk 6]
```diff
@@ -127,65 +127,158 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                      <div className="action-property-title">Output Files:</div>
+                      {this.state.actionResult.outputFiles ? (
+                        <div className="action-list">
+                          {this.state.actionResult.outputFiles.map((file) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:254]: nit: remote `PATH=`

**Review by bduffany — CHANGES_REQUESTED** [review]
1 nit, otherwise LGTM

File: app/format/format.tsx [hunk 7]
```diff
@@ -102,6 +102,12 @@ export function formatTimestampMillis(timestamp: number | Long) {
   return `${moment(+timestamp).format("MMMM Do, YYYY")} at ${moment(+timestamp).format("h:mm:ss a")}`;
 }
 
+export function formatTimestampSecondsNanos(seconds: number | Long, nanos: number | Long) {
+  return `${moment(+seconds * 1000).format("MMMM Do, YYYY")} at ${moment(+seconds * 1000).format(
```
Comment (siggisim) [inline app/format/format.tsx:106]: I think this could be simplified to 

```
export function formatTimestampSecondsNanos(seconds: number | Long, nanos: number | Long) {
  return `${formatTimestampMillis(seconds * 1000)}:${nanos.toString().substring(0, 2)}`;
}
```
also i'm curious if nanos always just has 2 digits? `.substring(0, 2)` seems a little brittle. can't there be like 1000000000 nanoseconds in a second?

might also be more convenient for the caller to just be able to pass in the timestamp object that has `seconds` and `nanos` as fields (it'd make the calling code much less verbose). i.e. i think you could make the type signature something like `formatTimestamp(timestamp: {seconds: number | Long, nanos: number | Long})` and use timestamp.seconds and timestamp.nanos below. that'd make the calling code much less verbose i.e.
```
format.formatTimestampSecondsNanos(this.state.actionResult.executionMetadata.outputUploadCompletedTimestamp)
```

Comment (pariparajuli) [reply app/format/format.tsx:106]: Yeah, I added 2 more digits from the nanos but past that, it usually just 0s

Comment (pariparajuli) [reply app/format/format.tsx]: also taking in the timestamp object like that in the parameter really simplified the code, ty!

Comment (siggisim) [reply app/format/format.tsx]: the timestamp object changes look great!

i'm not sure the substring changes are doing exactly what you'd expect. for example if one nanos value was 100 and the other was 1000 - both of these would show "10" using this method 

i feel like what we actually want is something like Math.floor(timestamp.nanos / 1000000)

File: app/invocation/invocation_action_card.tsx [hunk 8]
```diff
@@ -127,65 +127,137 @@ export default class ActionCardComponent extends React.Component<Props, State> {
...
+                        <div>None found.</div>
+                      )}
+                    </div>
+                    <div className="action-section">
+                      <div className="action-property-title">Output Directories:</div>
+                      {this.state.actionResult.outputDirectories.length ? (
+                        <div className="action-list">
+                          {this.state.actionResult.outputDirectories.map((dir) => (
+                            <div>
+                              <span className="prop-name">PATH=</span>
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:249]: nit: Remove this line too

**Review by siggisim — APPROVED** [review]
looks great!

**Review by bduffany — APPROVED** [review]
Nice!
